src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,137 @@
+    }
+
+    @Test
+    void register_loginLessThanSixSymbols_notOk() {
+        initialUser.setLogin("dummy");
+        assertThrows(RegistrationException.class, () -> registrationService.register(initialUser));
+    }
+
+    @Test
+    void register_loginHasSixOrMoreSymbols_ok() {
+        initialUser.setLogin("dummy@37");
+        initialUser.setAge(18);
+        initialUser.setPassword("password1234");
+        User userToCheck = registrationService.register(initialUser);
+        boolean expected = userToCheck.getLogin().length() > 5;
+        assertTrue(expected);
+    }
+
+    @Test
+    void register_loginIsEmptyString_notOk() {
+        initialUser.setLogin("");
+        assertThrows(RegistrationException.class, () -> registrationService.register(initialUser));
+    }
+
+    @Test
+    void register_ageLessThan18Years_notOk() {
+        initialUser.setAge(17);
+        assertThrows(RegistrationException.class, () -> registrationService.register(initialUser));
+    }
+
+    @Test
+    void register_ageIsNegativeNumber_notOk() {
+        initialUser.setAge(-12);
+        assertThrows(RegistrationException.class, () -> registrationService.register(initialUser));
+    }
+
+    @Test
+    void register_ageIsEqualTo18YearsOrMore_ok() {
+        initialUser.setLogin("dummy@37");
+        initialUser.setAge(18);
+        initialUser.setPassword("password1234");
+        User userToCheck = registrationService.register(initialUser);
+        boolean expected = userToCheck.getAge() > 17;
+        assertTrue(expected);
+    }
+
+    @Test
+    void register_ageIsNull_notOk() {
+        initialUser.setAge(null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(initialUser));
+    }
+
+    @Test
+    void register_passwordIsNull_notOk() {
+        initialUser.setPassword(null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(initialUser));
+    }
+
+    @Test
+    void register_passwordLessThanSixSymbols_notOk() {
+        initialUser.setPassword("dum");
+        assertThrows(RegistrationException.class, () -> registrationService.register(initialUser));
+    }
+
+    @Test
+    void register_passwordIsEmpty_notOk() {
+        initialUser.setPassword("");
+        assertThrows(RegistrationException.class, () -> registrationService.register(initialUser));
+    }
+
+    @Test
+    void register_userIsAlreadyExistInStorage_notAddToStorage() {
+        Storage.people.add(properToRegistUser);
+        int expected = Storage.people.size();
+        registrationService.register(properToRegistUser);
+        assertEquals(expected, Storage.people.size());
+    }
+
+    @Test
+    void register_userIsNotExistInStorage_addToStorage() {
+        int expected = Storage.people.size() + 1;
+        registrationService.register(properToRegistUser);
+        assertEquals(expected, Storage.people.size());
+    }
```

Comment:
combine with the positive test of registration, no need to have a separate test for that

Reply:
Done